packages/jest-resolve/src/__tests__/resolve.test.js

```diff
@@ -78,6 +78,19 @@ describe('findNodeModule', () => {
       paths: (nodePaths || []).concat(['/something']),
     });
   });
+
+  it('it cleans up the path if a query string is present', () => {
+    const newPath = Resolver.findNodeModule('./puppy.jpg?sizes=300w', {
+      basedir: '/',
+      browser: true,
+      extensions: ['js'],
+      moduleDirectory: ['node_modules'],
+      paths: ['/something'],
+      resolver: require.resolve('../__mocks__/userResolver'),
+    });
+
+    expect(userResolver.mock.calls[0][0]).toBe('./puppy.jpg');
```

Comment:
I would include the additional expectations they include in the previous test. Namely...

```javascript
expect(newPath).toBe('module');
expect(userResolver.mock.calls[0][0]).toBe('./test.jpg');
expect(userResolver.mock.calls[0][1]).toEqual({
  basedir: '/',
  browser: true,
  extensions: ['js'],
  moduleDirectory: ['node_modules'],
  paths: (nodePaths || []).concat(['/something']),
});
```